src/main/java/org/mskcc/oncokb/curation/web/rest/AuditResource.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    /**
+     * Fetches the last 100 change list for an entity class, if limit is passed fetches that many changes
+     *
+     * @param entityType  One of the AuditEntity types
+     * @param limit       number of records to return
+     * @param queryParams pagination parameters
+     * @return a list of entity audit events
+     */
+    @RequestMapping(value = "/audits/entity/changes", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<EntityAuditEvent>> getChanges(
+        @RequestParam(value = "entityType") String entityType,
+        @RequestParam(value = "limit") int limit,
+        @RequestParam MultiValueMap<String, String> queryParams,
+        UriComponentsBuilder uriBuilder
+    ) throws ClassNotFoundException {
+        log.debug("REST request to get a page of EntityAuditEvents");
+
+        Class<?> entityTypeToFetch = Class.forName(ENTITY_PACKAGE_PATH + entityType);
+        QueryBuilder jqlQuery = QueryBuilder.byClass(entityTypeToFetch).limit(limit);
+
+        List<CdoSnapshot> snapshots = javers.findSnapshots(jqlQuery.build());
+
+        List<EntityAuditEvent> auditEvents = new ArrayList<>();
+
+        snapshots.forEach(snapshot -> {
+            EntityAuditEvent event = EntityAuditEvent.fromJaversSnapshot(snapshot);
+            event.setEntityType(entityType);
+            auditEvents.add(event);
+        });
+
+        Page<EntityAuditEvent> page = new PageImpl<>(auditEvents);
+
+        HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(uriBuilder.queryParams(queryParams), page);
+
+        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
+    }
+
+    /**
+     * Fetches a previous version for an entity class, id and commit id
+     *
+     * @param entityType One of the AuditEntity types
+     * @param entityId   unique id in the selected entity table
+     * @param commitId   the commit id that audit table uses
+     * @return previous audit event. Return empty if nothing found.
+     */
+    @RequestMapping(
+        value = "/audits/entity/changes/version/previous",
+        method = RequestMethod.GET,
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<EntityAuditEvent> getPrevVersion(
+        @RequestParam(value = "entityType") String entityType,
+        @RequestParam(value = "entityId") Integer entityId,
+        @RequestParam(value = "commitId") Long commitId
+    ) throws ClassNotFoundException {
+        Class entityTypeToFetch = Class.forName(ENTITY_PACKAGE_PATH + entityType);
+
+        // Find the commit
+        QueryBuilder jqlQuery = QueryBuilder.byInstanceId(entityId, entityTypeToFetch).withCommitId(BigDecimal.valueOf(commitId)).limit(5);
+
+        // Since we use random commit id, we need to find the list of commits that prior to the commitId
+        List<CdoSnapshot> snapshots = javers.findSnapshots(jqlQuery.build());
```

Comment:
Are commit ids unique? I'm wondering why we need the limit 5. What is the difference if I set to limit 1?

Reply:
Yeah, makes sense. Must be a legacy code I forgot to update.